rdf-abac-core/src/test/java/io/telicent/jena/abac/attributes/TestAttributeParserEngine.java

```diff
@@ -0,0 +1,153 @@
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("(a & b | \"*\"");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("No RPAREN: [LPAREN:(]", exception.getMessage());
+    }
+
+    @Test
+    public void testReadExprUnary02() {
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("(a }");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("Expected RPAREN: [RBRACE:}]", exception.getMessage());
+    }
+
+    @Test
+    public void testReadExprUnary03() {
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("abc & {");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("No RBRACE: [LBRACE:{]", exception.getMessage());
+    }
+
+    @Test
+    public void testReadExprUnary04() {
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("a & { }");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("Expected WORD after: [LBRACE:{]", exception.getMessage());
+    }
+
+
+    @Test
+    public void testReadExprUnary05() {
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("{a & b");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("Expected RBRACE: [AMPERSAND:&]", exception.getMessage());
+    }
+
+    //TODO
+    // I get a NoSuchElementException on "{word", I would assume it should be Expected RBRACE or No RBRACE?
+    // Or is this correct behaviour?
+    @Test
+    public void testReadExprUnary06() {
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("{word");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("No RBRACE: [LBRACE:{]", exception.getMessage());
+    }
+
+    //TODO
+    // Trying to cover the lines 192-193, but it throws the "END" somewhere else
+    @Test
+    public void testReadExprRel01() {
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("{a} | ");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("END", exception.getMessage());
+    }
+
+    //TODO
+    // I can't figure out how to trigger the default case - seems to be covered by other exceptions"
+    @Test
```

Comment:
Again, I don't think this code is reachable. 

I think we should leave the code there though unlike the other code I've removed.